### ISSUE TITLE
IPNS not work with dht

### ISSUE THREAD

**Issue by way-zer** [ipfs/js-ipfs#3850]
- **Version**: ipfs-core@0.10.5
- **Platform**: windows x64 && node v16.1.0
- **Subsystem**: IPNS libp2p
#### Description:
When enable `libp2p.config.dht`, the `ipfs.name.resolve` fail.
And throw `Error: found ipns record that we couldn't convert to a value`
expected to work

#### Steps to reproduce the error:
```js
const last = require("it-last");
const assert = require("assert");
const {create} = require("ipfs-core")

;(async () => {
    const ipfs = await create({
        repo: './DAppAgent',
        config: {
            Bootstrap: [],
            Addresses: {
                Delegates: [],
            },
        },
        libp2p: {
            config: {
                dht: {enabled: true},
            },
        },
    })
    try {
        await ipfs.key.gen("test")
    } catch (e) {//exist
    }
    const r = await ipfs.name.publish("QmUNLLsPACCz1vLxQVkXqqLX5R1X345qqfHbsf67hvA3Nn", {key: "test"})
    const r2 = await last(ipfs.name.resolve(r.name))
    console.log(r, r2)
    assert(r.value === r2)
})()
```
#### May be the problem
https://github.com/libp2p/js-libp2p-kad-dht/blob/ee3b374cd72ae529d8a512df6cbe1ac38ffd82b8/src/content-fetching/index.js#L107
dht convert to record when put, but not use `Record.deserialize` when get

**Comment by way-zer** [comment]
fix after reinstall npm packages

**Comment by way-zer** [comment]
Now it throw `record requested for 12D3KooWNUXUJJGBHjNhXwpBEmSenD14ARepTtK6Q5g6X5Mkytn5 was not found in the network`

**Comment by way-zer** [comment]
selector be wrong when `records.length == 1`

**Comment by way-zer** [comment]
no reply？

**Comment by lidel** [comment]
@way-zer is my understanding correct, that you've found a bug when `records.length == 1` and were able to fix it?

DHT in js-ipfs is experimental and disabled by default, but if you have a working fix, do you mind opening a PR for a review?

**Comment by way-zer** [comment]
Without dht, how the IPNS works? I only see local store, pubsub and DHT. And pubsub is also experimental.

**Comment by achingbrain** [comment]
This was fixed in `libp2p-record@0.10.6`, you should be able to reinstall and/or upgrade ipfs to get the fix.

Please re-open if you're still seeing problems.
